cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/datastore/AttachmentManager.java

```diff
@@ -222,57 +220,55 @@ protected PreparedAttachment prepareAttachment(Attachment attachment, long lengt
         return pa;
     }
 
-    // take a set of attachments, and:
-    // * if attachment is saved, add it to the saved list
-    // * if attachment is not saved, prepare it, and add it to the prepared list
-    // this way, the attachments are sifted through ready to be added in the database for a given revision
-    protected PreparedAndSavedAttachments prepareAttachments(Collection<? extends Attachment> attachments) throws AttachmentException {
-        // actually a list of prepared or saved attachments
-        PreparedAndSavedAttachments preparedAndSavedAttachments = new PreparedAndSavedAttachments();
-
-        if (attachments == null || attachments.size() == 0) {
-            // nothing to do
-            return null;
+    /**
+     * Return a list of the existing attachments in the list passed in.
+     *
+     * @param attachments Attachments to search.
+     * @return List of attachments which already exist in the attachment store, or an empty list if none.
+     */
+    protected List<SavedAttachment> findExistingAttachments(Collection<? extends Attachment> attachments) {
```

Comment:
In addition, they need not be paired -- we could call the method to find Saved attachments within the transaction whereas the preparation does need to go outside the transaction.